source/openpower/keys.rst

```diff
@@ -0,0 +1,106 @@
+.. _keys:
+
+SSH Key Behavior on OpenPOWER
+=============================
+
+These are findings of how SSH Keys behave in various conditions.
+  
+Changing a Key From Inside an Instance
+--------------------------------------
+
+When we create our OpenPOWER OpenStack instance, we set up a key pair to create the machine with.
+
+.. note::
+
+  For the purposes of this documentation, this key pair will be known as the **Original Key Pair**.
+
+To allow for multiple SSH Keys to access this instance:
+  We need to add the contents of ``<our_key>.pub`` to ``~/.ssh/authorized_keys`` from the machine with the corresponding keyfile.
+  
+  To do so, we need to be certain we can ``ssh`` into our instance with:: 
+  
+      ~$ ssh <image_name>@<instance_IP_address>
+  
+  *Example:* ``~$ ssh centos@140.211.190.20``
+    
+  After we have tested that, we can ``Ctrl+D`` or ``logout`` out of the instance, and use ``cat`` and ``ssh`` to transfer ``<our_key>.pub`` to ``authorized_keys``, allowing multiple key pairs.
+  
+  - To **add** a key pair, retaining the original(s)::
+  
+      ~$ cat ~/.ssh/<our_key>.pub | ssh <image_name>@<instance_IP_address> "mkdir -p ~/.ssh && cat >> ~/.ssh/authorized_keys"
+      
+    *Example:* ``~$ cat ~/.ssh/id_rsa.pub | ssh debian@140.244.313.10 "mkdir...``
+    
+  - To **change** the key pair, removing all others::
+
+      ~$ cat ~/.ssh/<our_key>.pub | ssh <image_name>@<instance_IP_address> "mkdir -p ~/.ssh && cat > ~/.ssh/authorized_keys"
+      
+    The difference lies in the ``cat`` command, exchanging ``>>`` for ``>``
+    
+Changing a Key From OpenStack Web GUI
+-------------------------------------
+
+As far as in our knowledge, there is currently no way to edit the **Original Key Pair** designated to the instance.
+
+Not even through deleting the original key pair, creating a new one with the same name, and rebuilding the instance.
+
+Our best guess to the reason why is because when the instance is created, the **Original Key Pair** and its fingerprint are entered into the ``nova`` database, and there remain, uneditable, until its destruction.
+
+Note that we can use this knowledge to our advantage, however, as there is a fail-proof way to return the ``authorized_keys`` file to its original state by simply rebuilding the instance.
+
+Accessing an Instance After a Rebuild
+-------------------------------------
+
+As explained above, using SSH Key Pairs to access an instance after a rebuild will *only* work with the **Original Key Pair**.
+
+So... What do you do if you lost the original key?
+
+Accessing an Instance After a Rebuild ... *Without the Original Key*
+--------------------------------------------------------------------
+
+As of right now, the preferred method (still being researched) is administrator password injection.
```

Comment:
I think we should test this method out completely on our staging env before we ask customers to reach out to us in case their keys aren't working. Because this seems to be the only way ssh key resets works(correct me if I am wrong, @hodgesto)

We should test on staging:

1. How that setting would be reliably applied.
2. What is it's impact on machine and the access to machines for which we do not wish to reset the password?
3. What is it's impact on other dashboard users/projects?